packages/beacon-node/src/network/core/types.ts

```diff
@@ -1,17 +1,20 @@
 import {PeerScoreStatsDump} from "@chainsafe/libp2p-gossipsub/score";
 import {PublishOpts} from "@chainsafe/libp2p-gossipsub/types";
 import {routes} from "@lodestar/api";
-import {SpecJson} from "@lodestar/config";
+import {BlobScheduleEntry, SpecJson} from "@lodestar/config";
 import {LoggerNodeOpts} from "@lodestar/logger/node";
+import {ForkName} from "@lodestar/params";
 import {ResponseIncoming} from "@lodestar/reqresp";
-import {phase0} from "@lodestar/types";
+import {Epoch, phase0} from "@lodestar/types";
 import {PeerIdStr} from "../../util/peerId.js";
 import {NetworkOptions} from "../options.js";
 import {PeerAction, PeerScoreStats} from "../peers/index.js";
 import {OutgoingRequestArgs} from "../reqresp/types.js";
 import {CommitteeSubscription} from "../subnets/interface.js";
 
 export type MultiaddrStr = string;
+// Boundary of network subscription. We subscribe/unsubscribe during fork and blob schedule transitions
+export type SubscribeBoundary = {fork: ForkName; epoch?: Epoch} | ({fork: ForkName} & BlobScheduleEntry);
```

Comment:
suggest to make it more specific
```suggestion
export type SubscribeBoundary = {fork: ForkPreFulu; epoch: Epoch} | ({fork: ForkPostFulu} & BlobScheduleEntry);
```

then in both cases we have `epoch`, then for a lot of interfaces we only need epoch like `forkName2ForkDigestHex(epoch): ForkDigestHex`

here I suppose we only need to cache by epoch in out config as commented [here](https://github.com/ChainSafe/lodestar/pull/7954#discussion_r2144128838)

Reply:
> here I suppose we only need to cache by epoch in out config as commented https://github.com/ChainSafe/lodestar/pull/7954#discussion_r2144128838

We can cache epoch in genesis config but not here. 

Reason being `SubscribeBoundary` resides in `GossipTopic`, and we need fork info to do something like `getGossipSSZType`. Sometimes it is hard to get fork info from epoch because a lot of places don't have access to `config`. It would need some non-trivial refactoring for some places to have access to `config`

Reply:
> > here I suppose we only need to cache by epoch in out config as commented [#7954 (comment)](https://github.com/ChainSafe/lodestar/pull/7954#discussion_r2144128838)
> 
> We can cache epoch in genesis config but not here.
> 
> Reason being `SubscribeBoundary` resides in `GossipTopic`, and we need fork info to do something like `getGossipSSZType`. Sometimes it is hard to get fork info from epoch because a lot of places don't have access to `config`. It would need some non-trivial refactoring for some places to have access to `config`

There is also one issue with `SubscribeBoundary` being epoch only, is that when we are in the fulu fork, but before the first blob schedule, fork digest is calculated using ELECTRA_FORK_EPOCH [here](https://github.com/ethereum/consensus-specs/blob/1cc00b8adf001adb2bcdeda4e82838f2fe800da4/specs/fulu/beacon-chain.md?plain=1#L228), but the ssz types still use fulu's. So we need fork info to indicate which fork we are in to derive the ssz type, and epoch to indicate 1) where the boundary is 2)  calculate fork digest